doc/fluid/design/dist_train/mpi_enabled_design.md

```diff
@@ -0,0 +1,33 @@
+#MPI-enabled PaddlePaddle Design doc
+
+# Background
+Now, PaddlePaddle Fluid with Distribution has relatively large network bottleneck, We want to use RDMA and GPUDriect to improve and solve it, so we enabled the features to PaddlePaddle with the help of MPI.
```

Comment:
When we do distribute multi GPU training, the communication overhead between servers become the major bottleneck, because of the following reasons:

1. Must copy at least once from GPU to CPU memory so that the data can be ready to transfer. And for pserver side, copy data from CPU to GPU introduce more overhead.
1. GPU->CPU data transfer is 10 times slower than data transfer between GPUs or between PCIe devices.
1. TCP connections can not make full use of RDMA 100Gb devices.

Reply:
done